appserver/appclient/server/core/src/main/java/org/glassfish/appclient/server/core/jws/servedcontent/ASJarSigner.java

```diff
@@ -66,122 +70,110 @@
     /** property name optionally set by the admin in domain.xml to select an alias for signing */
     public static final String USER_SPECIFIED_ALIAS_PROPERTYNAME = "com.sun.aas.jws.signing.alias";
 
-    /** keystore type for JKS keystores */
-    private static final String JKS_KEYSTORE_TYPE_VALUE = "jks";
-
     /** default alias for signing if the admin does not specify one */
     private static final String DEFAULT_ALIAS_VALUE = "s1as";
 
-    private static final String DEFAULT_DIGEST_ALGORITHM = "SHA-1";
-    private static final String DEFAULT_SIGNATURE_ALGORITHM = "SHA1withRSA";
+    private static final String DEFAULT_DIGEST_ALGORITHM = "SHA-256";
+    private static final String DEFAULT_SIGNATURE_ALGORITHM = "SHA256withRSA";
 
     private static final SecuritySupport securitySupport = SecuritySupport.getDefaultInstance();
 
-//    /** user-specified signing alias */
-//    private final String userAlias; // = System.getProperty(USER_SPECIFIED_ALIAS_PROPERTYNAME);
-
     private static final StringManager localStrings = StringManager.getManager(ASJarSigner.class);
 
     private Logger logger;
 
     @Override
     public void postConstruct() {
-        logger =Logger.getLogger(JavaWebStartInfo.APPCLIENT_SERVER_MAIN_LOGGER, JavaWebStartInfo.APPCLIENT_SERVER_LOGMESSAGE_RESOURCE);
+        logger = Logger.getLogger(APPCLIENT_SERVER_MAIN_LOGGER, APPCLIENT_SERVER_LOGMESSAGE_RESOURCE);
     }
 
+
     /**
-     *Creates a signed jar from the specified unsigned jar.
-     *@param unsignedJar the unsigned JAR file
-     *@param signedJar the signed JAR to be created
-     *@param attrs additional attributes to be added to the JAR's manifest main section
-     *@return the elapsed time to sign the JAR (in milliseconds)
-     *@throws Exception getting the keystores from SSLUtils fails
+     * Creates a signed jar from the specified unsigned jar.
+     *
+     * @param unsignedJar the unsigned JAR file
+     * @param signedJar the signed JAR to be created
+     * @param attrs additional attributes to be added to the JAR's manifest main section
+     * @return the elapsed time to sign the JAR (in milliseconds)
+     * @throws Exception getting the keystores from SSLUtils fails
      */
-    public long signJar(final File unsignedJar, final File signedJar,
-        String alias, Attributes attrs) throws Exception {
-
-        final ZipOutputStream zout = new ZipOutputStream(
```

Comment:
BUG caused in 18f78207a194e756b49c1fb25bdcdf5f1b570b0e - JarSigner wraps the stream too

Reply:
Just for the record - the error message  from the JDK is not much helpful - "there is no current entry". But it is correct - there was no current entry for adding an entry into a zip file, which would be another zipfile.